src/modules/appointments/appointments.service.ts

```diff
@@ -116,16 +138,80 @@ export class AppointmentsService {
     return appointment;
   }
 
-  async findAll(): Promise<Appointment[]> {
-    try {
-      return await this.appointmentsRepository.find();
-    } catch (err) {
-      throw new HttpException(err, HttpStatus.BAD_REQUEST);
+  async findAll(
+    user: User,
+    roomId?: string,
+    patientId?: string,
+    doctorId?: string,
+  ): Promise<ScheduledAppointment[]> {
+    let doctor: string;
+    let patient: string;
+    const filter: SearchFilter = {};
+
+    if (
+      (user.role === Roles.ADMIN && doctorId) ||
+      (user.role === Roles.ADMIN && patientId)
+    ) {
+      if (doctorId) {
+        doctor = (await this.doctorsService.findOneById(doctorId)).id;
+        filter.doctorId = doctor;
+      }
+      if (patientId) {
+        patient = (await this.doctorsService.findOneById(doctorId)).id;
+        filter.patientId = patient;
+      }
+    }
+
+    if (user.role === Roles.DOCTOR) {
+      doctor = (
+        await this.doctorsService.findOneById(null, {
+          where: { userId: user.id },
+        })
```

Comment:
I think it's better to define separate methods in doctorsService like getByUserId, getById instead of generic one for all situations. It leads to non-obvious and complicated arguments.

Reply:
It needs refactoring all across the project, I think it should be one method findOne overloaded with several options like:
findOneById or findOneWithOptions